### ISSUE TITLE
Request: Batch subscribe/unsubscribe to mail lists

### ISSUE THREAD

**Issue by robinbeatty** [putyourlightson/craft-campaign#418]
Could we select multiple contacts and batch move them between lists – get this request a lot from clients. If we had a batch actions functionality, a really good feature would be to be for developers to be able to extend on batch actions to for their own use cases: e.g. set a custom field value to multiple contacts. Thanks

**Comment by bencroker** [comment]
Contacts don’t belong to mailing lists. Instead, each contact can be subscribed to (or unsubscribed from) multiple mailing lists, so “moving” contacts between lists is an inaccurate mental model. A simple way to subscribe/unsubscribe contacts to/from a mailing list is to first export the contacts from the contact index page, then import them from the import tab on the contacts page.

**Comment by robinbeatty** [comment]
Sure thing. I meant more like batch assign contacts to mail lists, which I think is a frequent use case. Currently, without importing/exporting the only way to do this is per contact on /admin/campaign/contacts/12345#tab02--mailing-lists which is no good if there are thousands of contacts. A batch assignment feature on the contacts list would be perfect + ability to extend on the batch assignment via custom code, e.g. assign custom field values

**Comment by bencroker** [comment]
Thanks for clarifying. I’ll keep this feature request open, but you’re the first person to ever ask for it so in the meantime it is low priority. 

You can achieve this programmatically via a custom module/plugin as follows.

```php
use putyourlightson\campaign\Campaign;

$contact = Campaign::$plugin->contacts->getContactByEmail('test@email.com');
$mailingList = Campaign::$plugin->mailingLists->getMailingListBySlug('new');
$subscriptionStatus = 'subscribed'; // or `unsubscribed`
Campaign::$plugin->mailingLists->addContactInteraction($contact, $mailingList, $subscriptionStatus);
```

**Comment by bencroker** [comment]
Added the ability to bulk subscribe and unsubscribe contacts to and from mailing lists in https://github.com/putyourlightson/craft-campaign/commit/6c2f8c1ab6ec27febf1e5d1cd52c5df213c1ba01, for the next release.

You can test this by running `composer require "putyourlightson/craft-campaign:dev-develop as 2.11.0"`.

**Comment by robinbeatty** [comment]
Wowzers, great 👍

**Comment by bencroker** [comment]
Released in [2.11.0](https://github.com/putyourlightson/craft-campaign/releases/tag/2.11.0).
